### PR TITLE
Support for LE certificates

### LINKED ISSUE #704: Let's Encrypt support
We should add first class Let's Encrypt support to Kontena Master that uses dns based verification and writes certificate to Kontena Vault. This allows to use LE certificates on any service that can handle certs from environment variables (not only Kontena Loadbalancer).

### PR DESCRIPTION
**Pull request by jnummelin**
This PR adds support for managing LE certificates with vault.

The docs explain the usage pattern but in nutshell we utilize the DNS challenge and do the verification and certificate request in two steps.

The registration step creates private key, if not imported into vault before, and associates it with a email address to be used as identity for LE.

WIP flag still as there's some cleanup and refactoring needed. 

ping @jakolehm @nevalla 

fixes #704

### CODE REVIEW

**Comment by jnummelin** [comment]
This does not yet automatically renew the certificates, that can be added in later commit or separate PR.

File: docs/using-kontena/certificates.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+By default this creates a new private key to be used with LE to identify the client.
+
+**Note:** If you have already registered with other means and have existing private key you wish to use you can import it into vault using specific name `LE_PRIVATE_KEY`
+```
+$ kontena vault write LE_PRIVATE_KEY "$(cat priv_key.pem)"
+```
+
+## Create domain authorization
+
+To be able to request certificates for a domain one must first prove that you are in charge of tht domain. For this Konante certificate management support DNS based authorization.
```
Comment (jakolehm) [inline docs/using-kontena/certificates.md]: Konante -> Kontena ? :)

File: server/Gemfile [hunk 2]
```diff
@@ -18,7 +18,9 @@ gem 'httpclient', '~> 2.3'
 gem 'authority'
 gem 'ruby_dig'
 gem 'rollbar', require: false
-gem "mongoid-enum"
+gem 'mongoid-enum'
+gem 'acme-client'
+gem 'json-jwt', '= 1.5.2'
```
Comment (jakolehm) [inline server/Gemfile]: why static version?

File: server/app/mutations/grid_certificates/authorize_domain.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    include Common
+    include Logging
+
+    required do
+      model :grid, class: Grid
+      string :domain
+    end
+
+
+    def validate
```
Comment (jakolehm) [inline server/app/mutations/grid_certificates/authorize_domain.rb]: does nothing, remove?

File: docs/using-kontena/certificates.md (hunk 1 above)
Comment (jnummelin) [inline docs/using-kontena/certificates.md]: Uups, I just re-branded our product. :D

File: server/Gemfile (hunk 2 above)
Comment (jnummelin) [inline server/Gemfile]: https://github.com/unixcharles/acme-client/issues/74

**Comment by jnummelin** [comment]
Rebased and cleaned

File: docs/using-kontena/certificates.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+Kontena integrates natively with [LetsEncrypt](letsencrypt.org) to bring easy to use certificate management for your services.
+
+Certificate management is integrated with Kontena [vault](vault.md) to handle certificates with proper security constraints.
+
+## Register for LE
+
+To be able to use LetsEncrypt one must first register as user. 
+```
+kontena certificate register <you@example.com>
```
Comment (jakolehm) [inline docs/using-kontena/certificates.md]: Document why we need email here

**Comment by jakolehm** [comment]
LGTM!

**Pull request merged**